scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,365 @@
+        theta_hat_b_i = theta_hat_b[indices]
+        percentiles[indices] = np.percentile(theta_hat_b_i, alpha_i)
+    return percentiles[()]  # return scalar instead of 0d array
+
+
+def _bca_interval(data, statistic, axis, alpha, theta_hat_b):
+    """Bias-corrected and accelerated interval """
+    # closely follows [2] "BCa Bootstrap CIs"
+    sample = data[0]  # only works with 1 sample statistics right now
+
+    # calculate z0_hat
+    theta_hat = statistic(sample, axis=axis)[..., None]
+    percentile = _percentile_of_score(theta_hat_b, theta_hat, axis=-1)
+    z0_hat = ndtri(percentile)
+
+    # calculate a_hat
+    jackknife_sample = _jackknife_resample(sample)
+    theta_hat_i = statistic(jackknife_sample, axis=-1)
+    theta_hat_dot = theta_hat_i.mean(axis=-1, keepdims=True)
+    num = ((theta_hat_dot - theta_hat_i)**3).sum(axis=-1)
+    den = 6*((theta_hat_dot - theta_hat_i)**2).sum(axis=-1)**(3/2)
+    a_hat = num / den
+
+    # calculate alpha_1, alpha_2
+    z_alpha = ndtri(alpha)
+    z_1alpha = -z_alpha
+    num1 = z0_hat + z_alpha
+    alpha_1 = ndtr(z0_hat + num1/(1 - a_hat*num1))
+    num2 = z0_hat + z_1alpha
+    alpha_2 = ndtr(z0_hat + num2/(1 - a_hat*num2))
+    return alpha_1, alpha_2
+
+
+def _bootstrap_ci_iv(data, statistic, axis, confidence_level, n_resamples,
+                     method, random_state):
+    """Input validation for bootstrap_ci"""
+
+    axis_int = int(axis)
+    if axis != axis_int:
+        raise ValueError("`axis` must be an integer.")
+
+    n_samples = 0
+    try:
+        n_samples = len(data)
+    except TypeError:
+        raise ValueError("`data` must be a sequence of samples.")
+
+    if n_samples == 0:
+        raise ValueError("`data` must contain at least one sample.")
+
+    data_iv = []
+    for sample in data:
+        sample = np.atleast_1d(sample)
+        if sample.shape[axis] <= 1:
+            raise ValueError("each sample in `data` must contain two or more "
+                             "observations along `axis`.")
+        sample = np.moveaxis(sample, axis_int, -1)
+        data_iv.append(sample)
+
+    # should we try: statistic(data, axis=axis) here?
+
+    confidence_level_float = float(confidence_level)
+
+    n_resamples_int = int(n_resamples)
+    if n_resamples != n_resamples_int or n_resamples_int <= 0:
+        raise ValueError("`n_resamples` must be a positive integer.")
+
+    methods = {'percentile', 'basic', 'bca'}
+    method = method.lower()
+    if method not in methods:
+        raise ValueError(f"`method` must be in {methods}")
+
+    message = "`method = 'BCa' is only available for one-sample statistics"
+    if n_samples > 1 and method == 'bca':
+        raise ValueError(message)
+
+    random_state = check_random_state(random_state)
+
+    return (data_iv, statistic, axis_int, confidence_level_float,
+            n_resamples_int, method, random_state)
+
+
+def bootstrap_ci(data, statistic, axis=0, confidence_level=0.95,
```

Comment:
This is a good candidate for using `*` to require that all arguments after `statistic` use an explicit keyword:
```suggestion
def bootstrap_ci(data, statistic, *, axis=0, confidence_level=0.95,
```
This use of `*` is the first guideline over in https://github.com/scipy/scipy/pull/13955, and those guidelines are still under discussion, so this is not a blocking request.

Reply:
We can change this if consensus is reached there, but I don't think I will apply that now.

Reply:
So far, the responses in https://github.com/scipy/scipy/pull/13955 to the recommendation for using `*` have been positive.  Is there is a reason to *not* use it?  (Note that, if we merge this PR without it, we'll have to add the use of `*` before 1.7 is released.  If the function is released without it, we can't add it later--that would  be a backwards incompatible API change.)